### DISCUSSION TITLE
Here a demo of my project with this super library (Text, Audio Spectrum, Clocks and more)

### DISCUSSION THREAD

**Discussion by FrenchLab47** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#153] [Show and tell]
Hello all,
I have not published the code yet but I propose you the first demos of my project made with this Library.
My panel (SwinGraf) can display messages, clocks, audio spectrum, animated Gifs, Weather, RSS Feeds, and direct drawings ... 

Thanks to mrfaptastic for this magnificent Library and for the answers to my questions at the very beginning of this experience. 

#### Description > https://github.com/FrenchLab47/SwinGraf
<img src="/FrenchLab47/SwinGraf/blob/main/images/SwinGraf.gif?raw=true" alt="SwinGraf.gif">

**Comment by mrcodetastic** [comment]
THAT
IS
AMAZING

Is that all running from one ESP32? Credit to you for publishing the source code for free (if you do). I'd be selling that as a commercial product personally and keeping the code to your self for the short term at least. Looks like a lot of hard work went into that!

**Reply by donnersm** [reply]
I would like to take a look at that code...it looks amazing...

**Reply by FrenchLab47** [reply]
Thanks !
These 2 videos are made with 3 panels superimposed  (and 3 x esp32).
I did this to display three different functions when video starts.

A large part of the code will be free. I plan to market certain things like PCBs and 3D prints. 

donnersm : He gives good result, but he would scare a real programmer. I can surely do the same thing with less line. But we'll talk about when I publish it in September.

**Reply by mrcodetastic** [reply]
I think you should sell it as a kit and provide the firmware, at least initally, as a binary or something. Get something back for your efforts. But anyway, looks great and good work!

**Reply by donnersm** [reply]
So i guess, all i can do is build a countdown clock with a hub75 to count down until September lol
Again looks great!

**Reply by FrenchLab47** [reply]
Thank you, it's a good idea!

**Reply by RX1900** [reply]
Very beautiful ，Can the code be shared？
